Examples/DissipativeIsing1d/ising1d.py

```diff
@@ -34,7 +34,7 @@
 
 # Observables
 obs_sx = nk.operator.LocalOperator(hi)
-obs_sy = nk.operator.LocalOperator(hi, dtype=complex)
+obs_sy = nk.operator.LocalOperator(hi, dtype=np.complex128)
```

Comment:
why do we need to pass numpy types now?

Reply:
From the numpy docs 

> Note that, above, we use the Python float object as a dtype. NumPy knows that int refers to np.int_, bool means np.bool_, that float is np.float_ and complex is np.complex_.

Reply:
I would therefore keep standard python data types, ```complex```, ```float``` etc also because it's not entirely transparent why ```LocalOperator``` would need a numpy data type

Reply:
Yeah, this was just me doing find and replace. Here we don't need it.

In the other place (the model), it's written here: https://netket.github.io/netket/preview/double-prec/docs/sharp-bits.html#nans-in-training-and-loss-of-precision . It's due to jax's handling of `float` vs `np.float64` documented [here](https://jax.readthedocs.io/en/latest/type_promotion.html?highlight=type-promotion).
As I write in the docs, `float` is weak so gets converted to `float32` as soon as he meets a float32, instead of keeping the precision.

Reply:
We don't really need those dtypes in the examples, as in this PR I am moving everything to strong double precision by default, but I am trying to give examples to users and push them to use `np.float**` by defaualt

Reply:
I understand but this seems to me an issue with Flax, essentially, since numpy has already moved towards standardizing its data type to be consistent with the standard python ones...

Reply:
for example in this table I can't see ```np_.float``` https://jax.readthedocs.io/en/latest/type_promotion.html?highlight=type-promotion so maybe it's also jax lagging a bit behind here?

Reply:
it's jax, not flax. it's not an issue, it's intended behaviour. 
>The promotion rules used by JAX for floating-point types are similar to those used by PyTorch.

>These differences are motivated by the fact that accelerator devices, such as GPUs and TPUs, either pay a significant performance penalty to use 64-bit floating point types (GPUs) or do not support 64-bit floating point types at all (TPUs). Classic NumPy’s promotion rules are too willing to overpromote to 64-bit types, which is problematic for a system designed to run on accelerators.

It's true that it's a bit surprising, which is why I am writing it in three places of the documentation.

We were not seeing those things before because we were handling parameters in a very unintended way, while now we are more consistent with jax.